### PR TITLE
Check for invalid inputs before applying new group size

### PR DESCRIPTION
**Pull request by JP1998**

### CODE REVIEW

File: pages/create.js [hunk 1]
```diff
@@ -163,12 +163,14 @@ class CreateTournamentForm extends React.Component {
...
     handleGroupSizeInput(input) {
         const newSize = input.target.value;
-        if (newSize <= this.state.groupAdvance) {
-            this.setState({
-                groupSize: newSize, groupAdvance: newSize - 1
-            });
-        } else {
-            this.setState({groupSize: newSize});
+        if (newSize !== undefined && newSize > 2) {
+            if (newSize <= this.state.groupAdvance) {
```
Comment (betanummeric) [inline pages/create.js]: Validating the input like following saves you from one indention level for the following default-case code:
```
if (newSize === undefined || newSize < 1) {
  return;
}
```

**Comment by JP1998** [comment]
I don't think so, as you could resolve those scenarios as follows:
 - group stage with only two players would be just like adding another playoff stage
 - group stage with only one player would make any team just advance
Also I didn't check the `groupAdvance` field, since it didn't contribute to the bug, and should thus (imo) only be checked by the backend. I'll implement this change nonetheless.

**Comment by betanummeric** [comment]
My bad. I confused things. Requiring a minimum of three *players per group* makes absolutely sense.

With checking the `groupAdvance` field I mean prohibiting values smaller than 0 or no value. This should be verified by the backend *and* the frontend.

File: pages/create.js [hunk 2]
```diff
@@ -178,7 +178,14 @@ class CreateTournamentForm extends React.Component {
     }
 
     handleGroupAdvanceInput(input) {
-        this.setState({groupAdvance: input.target.value});
+        const newAdvance = input.target.value;
+
+        if (newAdvance === undefined || newAdvance < 0 ||
```
Comment (betanummeric) [inline pages/create.js]: this could be `newAdvance <= 0`

Comment (JP1998) [reply pages/create.js]: Agreed, will change and force push, if you want me to

**Pull request merged**